### PR TITLE
Data collections for the front-page

### LINKED ISSUE #13: [Front Page] Data collections setup
Set up all data structures to hold the information for the different sections of the front page.

### PR DESCRIPTION
**Pull request by jdesiloniz**
Fixes #13 

This PR adds data collections for the new features to be added to the front page of the Scala Lang website. These are divided in different set of data for increased readabily and make it easier for contributors to update them:

- Fixed elements to be used in the front-page are specified as variables inside its markdown file (index.md). i.e.: scala IDEs, communities, gitter channels...
- Elements more subject to be changed are moved to YAML data files. i.e.: footer links, scala supporter/maintainer organizations...
- More complex and subject to change data elements are moved to custom collections. In this case, I've moved the main examples from the front page (previously hardcoded in partial layouts) to an specific collection.

Could you please review, @vejeta @dominv? Thanks!!

### CODE REVIEW

File: _data/common.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# SNS URLs
+githubUrl: http://github.com/scala/scala
+twitter: http://twitter.com/scala_lang
```
Comment (vejeta) [inline _data/common.yml]: Minor comment. These could be https URLs

File: _scala_items/1-seamless-java-interop.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+shortTitle: "Seamless Java Interop"
+shortDescription: "Scala runs on the JVM, so Java and Scala stacks can be freely mixed for totally seamless integration."
+scastieUrl: 
+---
+
+<div class="java-interop-wrapper" style="width: 500px">
+<figure class="code java-interop1" style="margin-top: 20px;">
```
Comment (vejeta) [inline _scala_items/1-seamless-java-interop.md]: Should the style properties be embedded into css classes? This is more a question for @israelperezglez

Comment (vejeta) [inline _scala_items/1-seamless-java-interop.md]: The same in other appearances of "style".

Comment (jdesiloniz) [inline _scala_items/1-seamless-java-interop.md]: Ah right, these are leftovers from the previous frontpage. These will disappear and be changed to a more appropiate structure when layouts are done by @israelperezglez.

**Comment by vejeta** [comment]
Besides minor comments, LGTM

**Pull request merged**